tests/manage/mcg/test_write_to_bucket.py

```diff
@@ -360,3 +369,46 @@ def test_write_to_bucket_rbd_cephfs(
                     awscli_pod_session, target_dir, full_object_path, mcg_obj
                 )
             )
+
+    @tier2
+    @bugzilla("2054074")
+    @skipif_ocs_version("<4.10")
+    @pytest.mark.polarion_id("OCS-4000")
+    def test_content_encoding_with_write(self, bucket_factory, mcg_obj_session):
+        """
+        Test s3 put object operation to see if the content-encoding is stored as object
+        metadata after put
+        """
+        # create bucket
+        bucket_name = bucket_factory()[0].name
+        logger.info(f"Bucket created {bucket_name}")
+
+        # create a random file and then zip it
+        filename = f"random-{uuid4().hex}"
+        with open(filename, "wb") as f:
+            f.write(os.urandom(1000))
+        zip_obj = ZipFile(f"{filename}.zip", "w")
+        zip_obj.write(f"{filename}", compress_type=zipfile.ZIP_DEFLATED)
+        zip_obj.close()
```

Comment:
You can change it to below and spare the use of close
with ZipFile(f"{filename}.zip", "w") as zip:
    zip.write(f"{filename}", compress_type=zipfile.ZIP_DEFLATED)